drivers/spi/spi_cc23x0.c

```diff
@@ -364,14 +391,19 @@ static int spi_cc23x0_transceive(const struct device *dev,
 	ret = spi_context_wait_for_completion(&data->ctx);
 	if (ret) {
 		LOG_ERR("SPI transfer failed (%d)", ret);
-		goto int_disable;
+		goto dma_suspend;
 	}
 
 	spi_context_update_tx(ctx, SPI_CC23_DFS, data->tx_len_left);
 	spi_context_update_rx(ctx, SPI_CC23_DFS, data->tx_len_left);
 
 	LOG_DBG("SPI transfer completed");
 
+dma_suspend:
+	ret = pm_device_runtime_put(cfg->dma_dev);
```

Comment:
i havent looked too closely, but if CONFIG_SPI_ASYNC is expected to work with this driver, this is probably wrong, because spi_context_wait_for_completion is mostly a NOP and doesn't wait if the transfer is async

Reply:
okay, true, nevermind, I guess I was used to drivers where those were just made into wrappers and they all went through the same code